src/compiler/checker.ts

```diff
@@ -12057,6 +12057,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (!links.type) {
             Debug.assertIsDefined(links.deferralParent);
             Debug.assertIsDefined(links.deferralConstituents);
+            if (links.deferralParent === neverType) {
+                const functionDecl = links.jsSyntheticRestParameterFunctionDeclaration!;
+                if (isFunctionExpressionOrArrowFunction(functionDecl)) {

```

Comment:
It kind of looks like you could check just `links.jsSyntheticRestParameterFunctionDeclaration`. Is the `neverType` actually relevant here?

Reply:
Yeah, you are right - this is just checking the same thing really and I can simplify it.